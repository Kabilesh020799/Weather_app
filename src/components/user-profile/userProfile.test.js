import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import UserProfile from './userProfile'; // Adjust the import path as necessary.

describe('UserProfile Component', () => {
  const userProps = {
    name: 'Kabzi',
    email: 'kabzi@gmail.com',
    picture: 'https://letsenhance.io/static/8f5e523ee6b2479e26ecc91b9c25261e/1015f/MainAfter.jpg'
  };
  it('renders user information correctly', () => {

    render(<UserProfile {...userProps} />);

    // Check if the name is rendered
    expect(screen.getByText(/Name:/i)).toBeInTheDocument();
    expect(screen.getByText('Kabzi')).toBeInTheDocument();

    // Check if the email is rendered
    expect(screen.getByText(/Email:/i)).toBeInTheDocument();
    expect(screen.getByText('kabzi@gmail.com')).toBeInTheDocument();

    // Check if the image is rendered with the correct src
    const image = document.querySelector('.user-profile-pic-img');
    expect(image).toHaveAttribute('src', 'https://letsenhance.io/static/8f5e523ee6b2479e26ecc91b9c25261e/1015f/MainAfter.jpg');
  });
  it('Snapshot Testing - UserProfile', () => {
    const { asFragment } = render(<UserProfile {...userProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
