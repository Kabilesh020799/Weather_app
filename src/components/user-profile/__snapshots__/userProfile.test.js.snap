// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile Component Snapshot Testing - UserProfile 1`] = `
<DocumentFragment>
  <div
    class="user_profile"
  >
    <div
      class="user_profile_pic"
    >
      <img
        alt="logo"
        class="user_profile_pic_img"
        id="user-profile-pic-img"
        src="https://letsenhance.io/static/8f5e523ee6b2479e26ecc91b9c25261e/1015f/MainAfter.jpg"
      />
    </div>
    <div
      class="user_profile_info"
    >
      <div
        class="user_profile_info_name"
      >
        <span>
          Name: 
        </span>
        <span>
          Kabzi
        </span>
      </div>
      <div
        class="user_profile_info_email"
      >
        <span>
          Email: 
        </span>
        <span>
          kabzi@gmail.com
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
